pkg/config/validation.go

```diff
@@ -0,0 +1,39 @@
+package config
+
+import (
+	"strings"
+
+	apimachineryvalidation "k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/utils/ptr"
+
+	configapi "sigs.k8s.io/lws/api/config/v1alpha1"
+)
+
+var (
+	internalCertManagementPath = field.NewPath("internalCertManagement")
+)
+
+func validate(c *configapi.Configuration) field.ErrorList {
+	var allErrs field.ErrorList
+	allErrs = append(allErrs, validateInternalCertManagement(c)...)
+	return allErrs
+}
+
+func validateInternalCertManagement(c *configapi.Configuration) field.ErrorList {
+	var allErrs field.ErrorList
+	if c.InternalCertManagement == nil || !ptr.Deref(c.InternalCertManagement.Enable, false) {
+		return allErrs
```

Comment:
This would return a nil error.

Maybe you should add an error message here that if certManagement is enabled we need the internalCertManager filed out.

Reply:
Thank you for review @kannon92.

I'm not sure I fully understand, If internal certificate management is disabled, shouldn't we return nil?. As far as I understand from the code block, if internal cert management is nil or false, we don't need to pursue further validation for the other fields.

Reply:
Yea that is true. Maybe we just return nil there to make it clear we don’t expect an error.

Reply:
I think this is fine as is. I was thinking we could validate if the settings are set if it’s enabled false so we bring awareness that they don’t be used